src/nendo_plugin_classify_core/plugin.py

```diff
@@ -171,7 +177,20 @@ def instruments(self, track: NendoTrack) -> dict:
         filtered_labels, _ = filter_predictions(
             predictions, settings.instrument_classes, threshold=0.05
         )
-        return {"instruments": filtered_labels[0]}
+        instruments = make_comma_separated_unique(filtered_labels)
+        return {"instruments": instruments}
+
+    @NendoAnalysisPlugin.plugin_data
+    def sfx(self, track: NendoTrack) -> dict:
+        """Compute the sfx of the given track."""
+        emb = self.embedding_model(get_signal(track))
+        predictions = self.sfx_model(emb)
```

Comment:
```suggestion
        # SR=16000 as recommended in the essentia documentation
        # https://essentia.upf.edu/reference/std_TensorflowPredictVGGish.html
        signal = es.MonoLoader(filename=track.resource.src, sampleRate=16000)()
        predictions = self.sfx_model(signal)
```

Reply:
The `TensorflowPredictVGGish` model wants to run on a mono signal of the track, not an embedding. With the changes I suggested, the detector ran smoothly. Thanks a ton for your contribution!